agent/util-scripts/gold/pbench-results-move/test-23.txt

```diff
@@ -15,11 +15,11 @@ Options:
   -m, --metadata TEXT            list of metadata keys to be sent on PUT.
                                  Option may need to be specified multiple
                                  times for multiple values. Format: key:value
-  --show-server TEXT             Display information about the pbench server
-                                 where the result(s) will be moved (Not
-                                 implemented)
+  --relay TEXT                   Specify a relay server host as
+                                 http[s]://host[:port]
+  --server TEXT                  Specify the Pbench Server 1.0 host as
+                                 https://host[:port]
```

Comment:
What does "obvious" mean? *We* know that `pbench-results-move` only works with "new" (authenticated?) servers. But; perhaps while qualifying in the documentation makes sense, maybe the command help summary should state that somehow and we should leave it out of the option help.

Reply:
I have two thoughts here.  The first regards the ".0":  if/when we release version 1.0.1 or 1.1, _I_ don't want to have to chase down all the places (especially in the documentation...) where we said "1.0" and update them.  And, that leads to the second thought:  if we don't mention a particular version, then we don't have that problem.

Nevertheless, you are correct that we need to serve our consumers.  But, I'm not convinced that a passing reference to a version number is sufficient for that.  How is a user expected to know whether they have their Agent configured to refer to a 0.69 Server or a "post-0.72" Server?  How do we expect them to select the appropriate command for the server that they are using?  (And, how long will it even matter?  That is, how long will it be before we retire `pbench-[copy|move]-results` and the Server that they work with?)

The best approach would be for the Agent command to print out something helpful in the failure case(s) where it is applicable.  That is, if the targeted Server doesn't answer, _then_ we could mention that maybe they are using the wrong server or wrong command.  That way, we aren't burdening the help text or the man page with verbiage which becomes less and less useful as time moves forward.

Reply:
Better diagnosis of errors at runtime is always good; diagnosing a failure to connect (or a missing local `id_rsa` file) as a server version mismatch is a stretch that may not be very helpful. At the least we should appropriately set expectations in the documentation.

Also, remember that when we release server 1.0 we start weaning users off the old protocol and for a "Pbench Agent 1.0" we won't support or even mention the 0.69 protocol. All this shall pass ...

Reply:
> diagnosing a failure to connect (or a missing local id_rsa file) as a server version mismatch is a stretch that may not be very helpful

It could be as simple as adding a line to the `ConnectionError` error message which says, "This command requires a post-v0.69 Pbench Server, is that what you have configured?"  🙂 

> we should appropriately set expectations in the documentation.

Documentation is what people read when they cannot understand the error messages....  😉 

> All this shall pass ...

Indeed...and I'm trying to make that passing as easy and quick as possible.  😆